src/engine.cairo

```diff
@@ -50,9 +50,19 @@ pub trait EngineTrait {
     fn has_flag(ref self: Engine, flag: ScriptFlags) -> bool;
     // Return the script since last OP_CODESEPARATOR
     fn sub_script(ref self: Engine) -> ByteArray;
+
+    // check the stack size before pushing data
+    fn check_stack_size(ref self: Engine) -> Result<(), felt252>;
 }
+pub const MAX_STACK_SIZE: u32 = 1000;
 
 pub impl EngineImpl of EngineTrait {
+    fn check_stack_size(ref self: Engine) -> Result<(), felt252> {
+        if self.dstack.len() >= MAX_STACK_SIZE {
```

Comment:
It seems the max stack size check should include the alt stack : https://github.com/btcsuite/btcd/blob/2b53ed198955ac40fe5b3ce4a854e9f5bfa68258/txscript/engine.go#L971

Reply:
so both astack and dstack combine, must be less than 1000 ?